### DISCUSSION TITLE
Can the SXA Redirect Plugin be used with the Next.js App set to REST?

### DISCUSSION THREAD

**Discussion by bewards** [Sitecore/jss#1456] [Help]
### Description

Since Sitecore Forms requires switching to REST layout service, would I not be able to use the SXA redirect plugin because the plugin is driven by GraphQL?

## Error Information
I've switched my local application over to FETCH=REST, restarted the services, and get the following error:
![image](https://user-images.githubusercontent.com/3927169/234706387-14a807d0-358a-483b-aae8-935ddf197b27.png)

I can still access and use the graphql edge playground and see a single test redirect I added even though I switched to REST..
![image](https://user-images.githubusercontent.com/3927169/234702832-1e266d22-b256-4fd9-a716-c090bb8f5692.png)

### Additional information

Current Version Info
------------------------
Sitecore v10.3
sitecore-jss-nextjs v21.0.8
sitecore-jss-forms v21.1.1
sitecore-jss-react-forms v21.1.1

**Comment by art-alexeyenko** [comment]
The FETCH env variable only specifies the fetch approach for Layout/Dictionary services. All of the middleware plugins will just use what they will (GraphQL in this case).
The error you get seems to related to missing jssAppName from the config. Please ensure that either the appName is not empty in your package.json, or JSS_APP_NAME is set in env.

Please also check if the issue still there when using the same version for all packages (i.e. 21.0.8)

**Comment by bewards** [comment]
The `generate-config.ts` file will pull the app name from the package.json file (config.appName) part of the npm run bootstrap script process (included in my pipeline). My jss app name is definitely coming through (also have the env var set within a .env.production file in the next.js app).

**Reply by art-alexeyenko** [reply]
I see. One other thing you could try is debug logging. If you set the env variable as DEBUG=sitecore-jss:redirects you should see extra logs from the redirects middleware - and that way you can ensure all of the parameters are being passed correctly.

**Reply by art-alexeyenko** [reply]
Were you able to find a fix? Please let me know if you need any more assistance.

**Reply by bewards** [reply]
Unfortunately I haven't had capacity to continue looking into this. It was suggested by other teams that we should try the Sitecore Embeddable Forms if our application is mainly driven by SSG/ISR, so that's where we are headed next.

**Reply by art-alexeyenko** [reply]
I see. If you will ever return to this - please reopen this thread and share your debug logs. They should shine some light on what exactly is it going wrong.
